### ISSUE TITLE
How can I figure out the (x,y) of each key points?

### ISSUE THREAD

**Issue by kirsten2000** [ildoonet/tf-pose-estimation#619]

**Comment by rdutta1999** [comment]
```
img = cv2.imread(f)
humans = e.inference(img, resize_to_default=(w > 0 and h > 0), upsample_size=resize_out_ratio)
```

`humans` here is a list of `tf.estimator.Human`. Each `tf.estimator.Human` contains info about each human and their pose points.
Now, each `tf.estimator.Human` has an attribute `body_parts`  = a dictionary that contains (x,y) and score for each partID, i.e. this contains the coordinates and score of each pose point. 
To retrieve these, you can simply write:-

```
for human in humans:
            for item in human.body_parts.items():   
                        // item is a tuple with item[0] being the ID and item[1] being class instance BodyPart
                        print("ID: ", item[0])
                        print("X Coord: item[1].x)
                        print("Y Coord: item[1].y)
                        print("Score : item[1].score)
```

**Comment by HansiIMW** [comment]
Hi, is there any info as to what the individual ID's are, like in this link from OpenPose by CMU? (https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/master/doc/output.md) 

Thanks for the info on looping btw, i receive the following output for just my face with this code: 
        for human in humans:
            for item in human.body_parts.items():
                print("ID: ", item[0])
                print("X Coord: ", item[1].x, "Y Coord: ", item[1].y)

ID:  0
X Coord:  0.3875 Y Coord:  0.3875
ID:  14
X Coord:  0.35 Y Coord:  0.3541666666666667
ID:  15
X Coord:  0.45625 Y Coord:  0.32083333333333336
ID:  16
X Coord:  0.325 Y Coord:  0.4083333333333333
ID:  17
X Coord:  0.565625 Y Coord:  0.3625

Since the coordinates are obviously normalized, do they also exist somewhere in non normalized form (just their pixel values)

**Comment by rdutta1999** [comment]
Yeah, if you would take a look at [tf_pose\common.py](https://github.com/ildoonet/tf-pose-estimation/blob/master/tf_pose/common.py), it mentions what each ID corresponds to which body part. 
```python
class CocoPart(Enum):
    Nose = 0
    Neck = 1
    RShoulder = 2
    RElbow = 3
    RWrist = 4
    LShoulder = 5
    LElbow = 6
    LWrist = 7
    RHip = 8
    RKnee = 9
    RAnkle = 10
    LHip = 11
    LKnee = 12
    LAnkle = 13
    REye = 14
    LEye = 15
    REar = 16
    LEar = 17
    Background = 18
```
It also describes which points are paired together to create the Pose Lines and the colour of these lines. 

As for your second question regarding where the non-normalized pixel values exist, you must first consider that each picture is initially resized to a particular dimension (the `resize` argument in [run.py](https://github.com/ildoonet/tf-pose-estimation/blob/master/run.py)). Then a `TfPoseEstimator` object is created, and its `inference()` function is called to get the `humans` list that I had mentioned previously:
`humans = e.inference(image, resize_to_default=(w > 0 and h > 0), upsample_size=args.resize_out_ratio)`

Following this to [estimator.py](https://github.com/ildoonet/tf-pose-estimation/blob/master/tf_pose/estimator.py), the `estimate_paf()` function is called:
```humans = PoseEstimator.estimate_paf(peaks, self.heatMat, self.pafMat)```.

 A snippet of the body of `estimate_paf()` is shown below:-

```python
human.body_parts[part_idx] = BodyPart(
                    '%d-%d' % (human_id, part_idx), part_idx,
                    float(pafprocess.get_part_x(c_idx)) / heat_mat.shape[1],
                    float(pafprocess.get_part_y(c_idx)) / heat_mat.shape[0],
                    pafprocess.get_part_score(c_idx)
                )
```
This is probably what you would need.
